File: javascript/easy/implementstrStr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if(needle.length > haystack.length) { return -1; }
    for(let i = 0; i < (haystack.length - needle.length + 1); i++) {
        let k = i, j = 0;
        while(j < needle.length) {
            if(needle[j] !== haystack[k]) {
                break;
            }
            j++; k++;
        }
        if(j == needle.length) {
            return i;
        }
    }
    return (haystack.length == 0 && needle.length == 0) ? 0 : -1;
};

File: ruby/easy/implement_str_str.rb
# @param {String} haystack
# @param {String} needle
# @return {Integer}
def str_str(haystack, needle)
    return -1 if needle.length > haystack.length
    for i in 0..(haystack.length - needle.length)
        k = i
        j = 0
        while j < needle.length
            break if needle[j] != haystack[k]
            j += 1
            k += 1
        end
        return i if j == needle.length
    end
    return -1
end
